core/domain/topic_domain.py

```diff
@@ -149,7 +149,7 @@ class TopicChange(change_domain.BaseChange):
         'user_id_attribute_names': []
     }, {
         'name': CMD_ADD_SUBTOPIC,
-        'required_attribute_names': ['title', 'subtopic_id'],
+        'required_attribute_names': ['title', 'subtopic_id', 'url_fragment'],
```

Comment:
This is added because the with the new validation, an empty `url_fragment` is invalid for the subtopic. The add_subtopic, create_default_subtopic functions both need to have a `url_fragment` arg, therefore the `ADD_SUBTOPIC` cmd also needs to have that attribute.

![image](https://user-images.githubusercontent.com/73544247/175980473-af1d7fb0-6503-42de-b737-059238a14b0f.png)

The frontend changes are required because we have updated the `ADD_SUBTOPIC` cmd, which is used in the frontend services.